vault/resource_generic_secret_test.go

```diff
@@ -47,7 +47,33 @@ func TestResourceGenericSecret_deleted(t *testing.T) {
 					}
 				},
 				Config: testResourceGenericSecret_initialConfig(path),
-				Check:  testResourceGenericSecret_initialCheck(path),
+				Check:  testResourceGenericSecret_initialCheck(path, false),
+			},
+		},
+	})
+}
+
+func TestResourceGenericSecret_deleteKeyMetadata(t *testing.T) {
+	path := acctest.RandomWithPrefix("secretsv2/test")
+	pathMetadata := "secretsv2/metadata/test"
+	resource.Test(t, resource.TestCase{
+		Providers: testProviders,
+		PreCheck:  func() { testAccPreCheck(t) },
+		Steps: []resource.TestStep{
+			{
+				Config: testResourceGenericSecret_initialConfig_v2(path),
+				Check:  testResourceGenericSecret_initialCheck(path, true),
+			},
+			{
+				PreConfig: func() {
+					client := testProvider.Meta().(*api.Client)
+					_, err := client.Logical().Delete(pathMetadata)
+					if err != nil {
+						t.Fatalf("unable to manually delete key metadata via the SDK: %s", err)
```

Comment:
Mentioning the here SDK could be confusing here, e.g. `Client` is a part of `api` and the provider relies on the `terraform-plugin-sdk`.

I don't understand the need for the `PreConfig`, I think what we really want here is `CheckDestroy`?

Reply:
You're right. Updated the error string to not reference SDK.

Reply:
@benashz added a `CheckDestroy` function in `b3fa092`